### DISCUSSION TITLE
Resolving turn and river

### DISCUSSION THREAD

**Discussion by mhluska** [b-inary/postflop-solver#36] [Q&A] [answered]
Let's say I specify two bet sizes for both players and solve from the flop. Now I'd like to add a third bet size on the turn only but without resolving the full game. How does one use this library to do that? 

I am currently creating a new `PostFlopGame` object with the previous game's `card_config` and `tree_config` copied over. The only difference being I update the pot size and stack sizes accordingly. And I update the `initial_state` to be `BoardState::Turn` as well.

Is this a reliable approach? I worry that it's not because the solver is losing knowledge of what happened on the flop. On the other hand, I _think_ I am copying the ranges over correctly so it should be fine:

```
let mut card_config = game.card_config().clone();
card_config.range = [
    Range::from_hands_weights(game.private_cards(0), game.weights(0)).unwrap(),
    Range::from_hands_weights(game.private_cards(1), game.weights(1)).unwrap(),
];
```

**Comment by b-inary** [comment, marked as answer]
I don't know if this approach is *reliable*, but I think it's typical. I don't think the solver loses knowledge of what happened on the flop, but strictly speaking, adding a third bet size on the turn may change the optimal strategy on the flop. But again, I think the change would be minimal and such a procedure is typical.
